resource_obmcs_core_instance_test.go

```diff
@@ -154,34 +161,109 @@ func (s *ResourceCoreInstanceTestSuite) TestAccResourceCoreInstance_basic() {
 				}`,
 				Check: resource.ComposeTestCheckFunc(
 					resource.TestCheckResourceAttr(s.ResourceName, "display_name", "-tf-instance"),
-					resource.TestCheckResourceAttr(vnicResourceName, "display_name", "-tf-vnic"),
-					resource.TestCheckResourceAttr(vnicResourceName, "skip_source_dest_check", "false"),
+					resource.TestCheckResourceAttr(s.ResourceName, "create_vnic_details.0.display_name", "-tf-vnic"),
+					resource.TestCheckResourceAttr(s.ResourceName, "create_vnic_details.0.skip_source_dest_check", "false"),
 					func(ts *terraform.State) (err error) {
 						newId, err := fromInstanceState(ts, s.ResourceName, "id")
-						if newId == instanceId {
-							return fmt.Errorf("Expected new instance ocid, got the same.")
+						if newId != instanceId {
+							return fmt.Errorf("Expected same instance ocid, got different.")
 						}
-						instanceId = newId
 						return err
 					},
 				),
 			},
-			// verify force new by creating an instance with different VNIC details.
+			// Adding create_vnic_details flags with default values should not lead to a change.
 			{
+				ImportState:       true,
+				ImportStateVerify: true,
 				Config: s.Config + `
 				resource "oci_core_instance" "t" {
 					availability_domain = "${data.oci_identity_availability_domains.ADs.availability_domains.0.name}"
 					compartment_id = "${var.compartment_id}"
 					image = "${data.oci_core_images.t.images.0.id}"
 					shape = "VM.Standard1.1"
 					display_name = "-tf-instance"
+					subnet_id = "${oci_core_subnet.t.id}"
 					metadata {
 						ssh_authorized_keys = "${var.ssh_public_key}"
 					}
 					create_vnic_details {
 						subnet_id = "${oci_core_subnet.t.id}"
 						display_name = "-tf-vnic"
+						skip_source_dest_check = false
 						assign_public_ip = true
+					}
+				}
+				data "oci_core_vnic_attachments" "t" {
+					compartment_id = "${var.compartment_id}"
+					instance_id = "${oci_core_instance.t.id}"
+				}
+				data "oci_core_vnic" "t" {
+					vnic_id = "${lookup(data.oci_core_vnic_attachments.t.vnic_attachments[0],"vnic_id")}"
+				}`,
+				ExpectNonEmptyPlan: false,
+				PlanOnly:           true,
+			},
+			// Update create_vnic_details
+			{
+				ImportState:       true,
+				ImportStateVerify: true,
+				Config: s.Config + `
+				resource "oci_core_instance" "t" {
+					availability_domain = "${data.oci_identity_availability_domains.ADs.availability_domains.0.name}"
+					compartment_id = "${var.compartment_id}"
+					image = "${data.oci_core_images.t.images.0.id}"
+					shape = "VM.Standard1.1"
+					display_name = "-tf-instance"
+					subnet_id = "${oci_core_subnet.t.id}"
+					metadata {
+						ssh_authorized_keys = "${var.ssh_public_key}"
```

Comment:
tangential not required change, we dont need to pass public keys to these test instances, can just pass `metadata {}`

Reply:
I guess we are generally testing typical params of an instance though so 
![ignoreme](https://user-images.githubusercontent.com/1305019/31473604-8142129c-aeaa-11e7-87e5-f73dd4dbecfb.jpg)